src/Controller/Component/FlashComponent.php

```diff
@@ -40,11 +42,72 @@ class FlashComponent extends Component
     protected $_defaultConfig = [
         'key' => 'flash',
         'element' => 'default',
+        'type' => 'default',
         'params' => [],
         'clear' => false,
         'duplicate' => true,
     ];
 
+    /**
+     * Called after the Controller::beforeRender(), after the view class is loaded, and before the
+     * Controller::render()
+     *
+     * @param \Cake\Event\EventInterface $event Event.
+     * @return \Cake\Http\Response|null
+     */
+    public function beforeRender(EventInterface $event)
+    {
+        /** @var \Cake\Controller\Controller $controller */
+        $controller = $event->getSubject();
+
+        if (!$controller->getRequest()->is('ajax')) {
+            return null;
+        }
+
+        if (
+            !in_array('yes', array_map( // case insensitive
+                'strtolower',
+                $controller->getRequest()->getHeader('X-Get-Flash')
```

Comment:
Could these extensions be done in a sub-class? Without other tools that support this header and its response header we end up with a bespoke implementation that nothing else supports.

Reply:
Yep - just wanted to canvas if there was interest in this in core.